sulong/projects/com.oracle.truffle.llvm.runtime/src/com/oracle/truffle/llvm/runtime/debug/debugexpr/nodes/DebugExprNodeFactory.java

```diff
@@ -0,0 +1,309 @@
+package com.oracle.truffle.llvm.runtime.debug.debugexpr.nodes;
+
+import java.util.List;
+
+import com.oracle.truffle.api.Scope;
+import com.oracle.truffle.api.TruffleLanguage.ContextReference;
+import com.oracle.truffle.llvm.runtime.ArithmeticOperation;
+import com.oracle.truffle.llvm.runtime.CompareOperator;
+import com.oracle.truffle.llvm.runtime.LLVMContext;
+import com.oracle.truffle.llvm.runtime.debug.debugexpr.nodes.DebugExprBitFlipNodeFactory.BitFlipNodeGen;
+import com.oracle.truffle.llvm.runtime.debug.debugexpr.nodes.DebugExprNotNode.NotNode;
+import com.oracle.truffle.llvm.runtime.debug.debugexpr.nodes.DebugExprNotNodeFactory.NotNodeGen;
+import com.oracle.truffle.llvm.runtime.debug.debugexpr.parser.DebugExprException;
+import com.oracle.truffle.llvm.runtime.debug.debugexpr.parser.DebugExprType;
+import com.oracle.truffle.llvm.runtime.debug.debugexpr.parser.Parser;
+import com.oracle.truffle.llvm.runtime.nodes.api.LLVMExpressionNode;
+import com.oracle.truffle.llvm.runtime.types.PrimitiveType;
+
+public final class DebugExprNodeFactory {
+    private ContextReference<LLVMContext> contextReference;
```

Comment:
I think it's fine. As far as I can tell, this object only lives for a single parse, so you can see this field as a kind of "cache" for efficiency. `LLVMLanguge.getLLVMContextReference` always has to look up a thread local field.

I would even go so far and store the `LLVMContext` here directly. In AST nodes that's not allowed, but here it would be fine, since this object is not persistent anyway.